src/stores/UserProfilesStore.ts

```diff
@@ -96,17 +135,31 @@ export class UserProfilesStore {
         return profile;
     }
 
+    public flush(): void {
+        this.profiles = new LruCache<string, IMatrixProfile | null>(cacheSize);
+        this.profileLookupErrors = new LruCache<string, MatrixError>(cacheSize);
+        this.knownProfiles = new LruCache<string, IMatrixProfile | null>(cacheSize);
+    }
+
     /**
      * Looks up a user profile via API.
      *
      * @param userId - User Id for which the profile should be fetched for
      * @returns The profile information or null on errors
      */
-    private async fetchProfileFromApi(userId: string): Promise<IMatrixProfile | null> {
+    private async fetchProfileFromApi(userId: string, options?: GetOptions): Promise<IMatrixProfile | null> {
         try {
             return (await this.client.getProfileInfo(userId)) ?? null;
```

Comment:
Maybe I'm missing it, but should any `profileLookupErrors` for this userId be cleared here too?